### PR TITLE
ts: Use const enum whenever possible

### PR DESCRIPTION
**Pull request by ikokostya**
Documentation: http://www.typescriptlang.org/docs/handbook/enums.html

### CODE REVIEW

File: typescript.md [hunk 1]
```diff
@@ -155,3 +156,12 @@ from JS).
   ```
 
 [&#8593; back to TOC](#table-of-contents)
+
+## Enums
+
+* Use `const` enums whenever possible.
+
+  > Explanation: const enums can only use constant enum expressions and unlike regular enums they
+  > are completely removed during compilation. Const enum members are inlined at use sites.
```
Comment (dfilatov) [inline typescript.md]: "use sites" ?

Comment (ikokostya) [inline typescript.md]: This is from [handbook](https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/Enums.md). "use sites" - в месте использования.

**Comment by dmikis** [comment]
There's a drawback in using constant enums exclusively: basic ones also provides 'enum-to-string' functionality out of the box:

```typescript
enum Foo {
    BAR,
    BAZ
}

Foo[Foo.BAR]; // => 'BAR'
```

If we all agree that necessity in that falls under 'not possible to use const enum' situation then I'm OK with the proposed change.

**Comment by ikokostya** [comment]
This is good point. Need update pull request and add more detailed description.

**Comment by ikokostya** [comment]
:up:

**Pull request merged**